Tests/AsyncHTTPClientTests/AsyncAwaitEndToEndTests.swift

```diff
@@ -745,17 +744,32 @@ final class AsyncAwaitEndToEndTests: XCTestCase {
             XCTAssertEqual(response.version, .http2)
         }
     }
-}
-
-extension AsyncSequence where Element == ByteBuffer {
-    func collect() async rethrows -> ByteBuffer {
-        try await self.reduce(into: ByteBuffer()) { accumulatingBuffer, nextBuffer in
-            var nextBuffer = nextBuffer
-            accumulatingBuffer.writeBuffer(&nextBuffer)
+    
+    func testSimpleContentLengthError() {
+        guard #available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *) else { return }
+        XCTAsyncTest {
+            let bin = HTTPBin(.http2(compress: false)) { _ in HTTPEchoHandler() }
+            defer { XCTAssertNoThrow(try bin.shutdown()) }
+            let client = makeDefaultHTTPClient()
+            defer { XCTAssertNoThrow(try client.syncShutdown()) }
+            let logger = Logger(label: "HTTPClient", factory: StreamLogHandler.standardOutput(label:))
+            var request = HTTPClientRequest(url: "https://localhost:\(bin.port)/")
+            request.method = .GET
+            request.body = .bytes(ByteBuffer(string: "1234"))
```

Comment:
Hrm, I think this muddies the waters of the test somewhat. I don't think this test would fail if you took away the content-length checks, because the actual `collect` call will still end up violating the content-length.

What you want is for the HTTPbin server to send you back a content-length that's too long and no body at all. That should still throw this error.